src/PIL/ImageFilter.py

```diff
@@ -371,20 +379,81 @@ def generate(cls, size, callback, channels=3, target_mode=None):
                          three color channels. Will be called ``size**3``
                          times with values from 0.0 to 1.0 and should return
                          a tuple with ``channels`` elements.
-        :param channels: Passed to the constructor.
-        :param target_mode: Passed to the constructor.
+        :param channels: The number of channels which should return callback.
+        :param target_mode: Passed to the constructor of the resulting
+                            lookup table.
         """
         size1D, size2D, size3D = cls._check_size(size)
-        table = []
+        if channels not in (3, 4):
+            raise ValueError("Only 3 or 4 output channels are supported")
+
+        table = [0] * (size1D * size2D * size3D * channels)
+        idx_out = 0
         for b in range(size3D):
             for g in range(size2D):
                 for r in range(size1D):
-                    table.append(callback(
-                        r / float(size1D-1),
-                        g / float(size2D-1),
-                        b / float(size3D-1)))
-
-        return cls((size1D, size2D, size3D), table, channels, target_mode)
+                    table[idx_out:idx_out + channels] = callback(
+                        r / (size1D-1), g / (size2D-1), b / (size3D-1))
+                    idx_out += channels
+
+        return cls((size1D, size2D, size3D), table, channels=channels,
+                   target_mode=target_mode, _copy_table=False)
+
+    def transform(self, callback, with_normals=False, channels=None,
+                  target_mode=None):
+        """Transforms the table values using provided callback and returns
+        a new LUT with altered values.
+
+        :param callback: A function which takes old lookup table values
+                         and returns a new. The number of arguments which
```

Comment:
'and returns a new'. And returns a new set of values?

Reply:
Fixed. Thanks!